cpp/include/cuspatial/experimental/detail/point_linestring_distance.cuh

```diff
@@ -83,34 +105,64 @@ void __global__ pairwise_point_linestring_distance(Cart2dItA points_first,
   for (auto idx = threadIdx.x + blockIdx.x * blockDim.x;
        idx < std::distance(linestring_points_first, thrust::prev(linestring_points_last));
        idx += gridDim.x * blockDim.x) {
-    auto offsets_iter =
-      thrust::upper_bound(thrust::seq, linestring_offsets_first, linestring_offsets_last, idx);
+    auto linestring_offsets_iter = thrust::upper_bound(
+      thrust::seq, linestring_part_offsets_first, linestring_part_offsets_last, idx);
```

Comment:
If you pass a counting_iterator for `linestring_part_offsets_first` and `last`, does this get compiled away?  This would be a good thing to verify...  If the counting iterator starts at zero, it should go away completely. If not, it should compile down to a single add, right?

Reply:
I don't think it's getting optmized away: https://godbolt.org/z/YfKanr9eP, and also, the iteartor that may use counting iterator is `linestring_geometry_offsets_first`.